pkg/provider/userinfo.go

```diff
@@ -52,7 +52,13 @@ func UserInfoGetterFactory(userProjectMapper ProjectMemberMapper) (UserInfoGette
 				return nil, err
 			}
 		} else {
-			groups.Insert(user.Spec.Groups...)
+			for _, group := range user.Spec.Groups {
+				groupName := group
+				if projectID != "" {
+					groupName += "-" + projectID
```

Comment:
Very minor and personal nit, but can we make this

```suggestion
					groupName += fmt.Sprintf("-%s", projectID)
```
?